extensions-core/kafka-indexing-service/src/main/java/org/apache/druid/indexing/kafka/supervisor/KafkaSupervisorSpec.java

```diff
@@ -131,6 +120,13 @@ public Supervisor createSupervisor()
     );
   }
 
+  @Override
+  @JsonProperty
+  public DataSchema getDataSchema()
+  {
+    return super.getDataSchema();
```

Comment:
Seems unnecessary.

Reply:
And `getDataSchema()` should be deprecated.

Reply:
removed from subclass and deprecated in base class